packages/@aws-cdk/aws-logs/lib/query-definition.ts

```diff
@@ -16,18 +16,21 @@ export interface QueryStringProps {
   readonly fields?: string[];
 
   /**
-  * Extracts data from a log field and creates one or more ephemeral fields that you can process further in the query.
+  * One or more statements for parsing data from a log field and creating ephemeral fields that can
+  * be processed further in the query. Each statement provided generates a separate parse line in
+  * the query string.
   *
   * @default - no parse in QueryString
   */
-  readonly parse?: string;
+  readonly parse?: string | string[];
```

Comment:
We can't use union types with jsii so this won't work/ Let's deprecate this prop and create a new on that's an optional `string[]`. Actually, can we also make this more strongly typed? If not, that's fine, but it would be nice to tighten the contract if we're deprecating props anyway.

Reply:
Hey @TheRealAmazonKendra -- I'm not the most familiar with typescript or CDK so just checking if I understand: by more strongly typed do you mean something like adding a specific "ParseStatement" type which has "inputAttribute", "parsePattern" and "outputAttributes" as members? I could do this to ensure the generated line is in the correct format, and validation could even be added to catch user errors early at build time (e.g. when number of "*" in the parsePattern != length of outputAttributes).

However, I'm not sure if this would be too heavyweight (I would have assumed so, even though I don't mind spending the time to add it in), so interested in your thoughts?

Reply:
I made the change to support string[] instead -- my feeling is that the above is too heavyweight anyway, but let me know :)